FluidNC/src/Control.cpp

```diff
@@ -22,6 +22,8 @@ void Control::init() {
 }
 
 void Control::group(Configuration::HandlerBase& handler) {
+    handler.item("enable_dual_mode_reset_pin", _dual_mode_reset); // reset button becomes a "feedhold button" if active state is "Cycle" or "Jog"
+    handler.item("enable_dual_mode_cycle_start_pin", _dual_mode_cycle_start); // cycle_start button becomes an "unlock button" if active state is "Alarm"
```

Comment:
We need better names.  "enable_dual_mode_reset_pin" is both too wordy and insufficiently descriptive - "dual_mode" doesn't really tell you much unless you already know what it means.  Maybe "reset_does_feedhold" and "cycle_start_does_unlock".  I'm not thrilled by those names but they are better than "enable_dual_mode..."

Reply:
Several people have asked for toggle style button like feed and cycle start.

The "enable" is not needed in the name. Maybe the names are the 2 functions like "feedhold_reset_pin".

We should also have a way to report these in the status report.

Reply:
I suppose the reporting for a reset/feedhold pin, could just report RH when active. The reporting is just done for human eyes, typically during troubleshooting.

Reply:
Sorry, I made the incorrect assumption this was defining a new pin. We have [spec for config item names](https://github.com/bdring/FluidNC/wiki/FluidNC-Config-File-Overview#names-and-units). "_pin" at the end would suggest the value should be of type Pin.

I wonder if that would be a better way to do it.

Reply:
Thank you both for your comments! By the way, I am now a sponsor of FluidNC. Very thrilled with the work you guys are doing!

@MitchBradley : Yeah... the names are pretty bad 😬... I think I like "reset_pin_does_feedhold". or "reset_pin_also_holds"? English is a second language, so this is not easy. And us french (Canadian in my case) are often wordy :-)

@bdring Very good point that it should not end with "_pin"! My first iteration was actually done this way, using 2 extra pin variables, but I felt it was a bit redundant to have both a reset_pin and a "reset_and_feedhold_pin". Also, I didn't want to add to the complexity of the realtime behaviour of "reset" (which is done via MotionControl.cpp I think while homing for instance). By using a "reset_also_holds" type of parameter, I was able to keep the changes to a minimum and just rely on existing functions and variables (rtReset, rtFeedhold...).

Reply:
Thank you. 

Here is a [discord invite link](https://discord.gg/XpytXt76). If you join Discord, please tell me your users name and I will add you to the "sponsor" group.

I looked at the code and agree with the mess new pin types create in the main code.

I think a standardized way would be to use the pin name first, then the added feature.

 reset_does_feedhold
 cycle_start_does_feedhold
 cycle_start_does_reset